020-quarkus-http-non-application-endpoints/src/test/java/io/quarkus/qe/non_application/endpoint/CommonNonAppEndpoint.java

```diff
@@ -0,0 +1,69 @@
+package io.quarkus.qe.non_application.endpoint;
+
+import static io.restassured.RestAssured.given;
+
+
+import java.net.InetAddress;
+import java.net.ServerSocket;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+import javax.net.ServerSocketFactory;
+
+import org.jboss.logging.Logger;
+
+import io.quarkus.test.common.QuarkusTestResourceLifecycleManager;
+import io.restassured.RestAssured;
+import io.restassured.builder.RequestSpecBuilder;
+import io.restassured.specification.RequestSpecification;
+
+public abstract class CommonNonAppEndpoint implements QuarkusTestResourceLifecycleManager {
+    private static final Logger LOG = Logger.getLogger(CommonNonAppEndpoint.class);
+
+    protected static final String ROOT_BASE_PATH = "/api/";
+    private RequestSpecification request;
+    private RequestSpecBuilder spec;
+    private static final String DEFAULT_PORT = "8081";
+
+    protected final List<String> nonAppEndpoints = Arrays.asList(
+            "/openapi", "/metrics/base", "/metrics/application",
+            "/metrics/vendor", "/metrics", "/health/group", "/health/well", "/health/ready",
+            "/health/live", "/health");
+
+    @Override
+    public Map<String, String> start() {
+        return null;
+    }
+
+    @Override
+    public void stop() {
+    }
+
+    protected void givenBasePath(String basePath) {
+        RestAssured.enableLoggingOfRequestAndResponseIfValidationFails();
+        this.spec = new RequestSpecBuilder();
+        this.spec.setBasePath(basePath);
+    }
+
+    protected void whenMakeRequestOverNonAppEndpoints() {
+        this.request = spec.build();
+    }
+
+    protected void thenStatusCodeShouldBe(int status) {
+        for (String endpoint : nonAppEndpoints) {
+            given().spec(request).log().uri().expect().statusCode(status).when().get(endpoint);
+        }
+    }
+
+    public static String getAvailablePort() {
```

Comment:
I would use the official way from Quarkus to select a random port: https://quarkus.io/guides/getting-started-testing#controlling-the-test-port

Reply:
**Note:**  random port only works if is defined on your `application.properties`
```
quarkus.http.test-port=0
```